src/averageAges.js

```diff
@@ -14,12 +14,27 @@
  *
  * @return {number}
  */
+const SEX_MALE = 'm';
+const SEX_FEMALE = 'f';
+const CENTURY_VALUE = 100;
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const filteredPeopleByMan = people
+    .filter(({ sex }) => sex === SEX_MALE);
+
+  const filteredPeopleByDiedMan = people
+    .filter(({ sex, died }) => {
+      return sex === SEX_MALE && Math.ceil(died / CENTURY_VALUE) === century;
+    });
+
+  const men = century
```

Comment:
Now everything is written correctly, but if `century` is `false`, then you are doing unnecessary filtering. It is better to first check if this value is `true` and then if it is necessary to do a new filtering.